### PR TITLE
Handling of onAvailableDrillTargetsReceived 

### PR DESCRIPTION
**Pull request by ivanmjartan**
---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4572/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6193/ : FAILURE in 6m 17s

**Comment by ivanmjartan** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4573/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6194/ : SUCCESS in 16m 51s

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/DefaultDashboardInsight.tsx [hunk 1]
```diff
@@ -65,6 +68,15 @@ export const DefaultDashboardInsight = (props: IDashboardInsightProps): JSX.Elem
     const [isVisualizationLoading, setIsVisualizationLoading] = useState(false);
     const [visualizationError, setVisualizationError] = useState<GoodDataSdkError | undefined>();
 
+    const handlePushData = useCallback(
+        (data: IPushData): void => {
+            if (onAvailableDrillTargetsReceived && data?.availableDrillTargets) {
+                onAvailableDrillTargetsReceived(data?.availableDrillTargets);
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/DefaultDashboardInsight.tsx:74]: Nitpick: Optional chaining should not be necessary inside the condition

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardDrillTargets.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import { useCallback, useState } from "react";
+import {
+    addDrillTargets,
+    IDrillTargets,
+    selectDrillTargetsByWidgetRef,
+    useDashboardDispatch,
+    useDashboardSelector,
+} from "../../../../model";
+
+export interface UseDashboardDrillTargetsProps {
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardDrillTargets.ts:14]: Missing alpha / internal annotation.

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardDrillTargets.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    widget: IInsightWidget;
+}
+
+interface UseDashboardDrillTargetsResult {
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardDrillTargets.ts:18]: Missing alpha / internal annotation & docs.

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardDrillTargets.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                !isEqual(drillTargets?.availableDrillTargets, availableDrillTargets)
+            ) {
+                dispatch(addDrillTargets(widget.ref, availableDrillTargets));
+            }
+        },
+        [widget, drillTargets],
+    );
+
+    return {
+        drillTargets,
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardDrillTargets.ts:48]: Do we want to return full drillTargets interface? Maybe availableDrillTargets is sufficient? Same question for the second hook.

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardInsightDrills.ts [hunk 5]
```diff
@@ -33,12 +25,12 @@ export interface UseDashboardInsightDrillsProps {
     widget: IInsightWidget;
     insight: IInsight;
     drillableItems?: Array<IDrillableItem | IHeaderPredicate>;
+    drillTargets: IDrillTargets | undefined;
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardInsightDrills.ts:28]: Nitpick: `drillTargets?: IDrillTargets`

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/DefaultDashboardInsight.tsx (hunk 1 above)
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/DefaultDashboardInsight.tsx:74]: Done

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardInsightDrills.ts (hunk 5 above)
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardInsightDrills.ts]: Done

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardDrillTargets.ts (hunk 4 above)
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardDrillTargets.ts]: Done

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardDrillTargets.ts (hunk 2 above)
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardDrillTargets.ts:13]: Done

File: libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardDrillTargets.ts (hunk 3 above)
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/presentation/widget/insight/DefaultDashboardInsight/useDashboardDrillTargets.ts:17]: Done

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4579/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6200/ : SUCCESS in 17m 47s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by kandl** [comment]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1568

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1301/ : SUCCESS in 1m 03s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1577/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/351/ : SUCCESS in 14m 52s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6206/ : SUCCESS in 21m 49s

**Pull request merged**